### PR TITLE
New deck command

### LINKED ISSUE #53: New deck command
Creates a new deck 
Command new deck

### LINKED ISSUE #54: Remove deck command
Removes a deck
Command: remove <index>

### PR DESCRIPTION
**Pull request by MelanieNPS**
Add NewDeckCommand.java and its parser
Command Word: new <deck_name>

Add RemoveDeckCommand.java and its parser
Command Word: remove <index>

### CODE REVIEW

File: src/main/java/seedu/address/model/deck/Deck.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    @Override
+    public int hashCode() {
+        return Objects.hash(deckName, cards);
+    }
+
+    @Override
+    public String toString() {
+        final StringBuilder builder = new StringBuilder();
+        builder.append(getDeckName())
+                .append(" Cards: ");
```
Comment (geoboom) [inline src/main/java/seedu/address/model/deck/Deck.java:71]: Hmm indentation looks a bit off - not sure if it's just GitHub.
Did you `CMD/CTRL+ALT+L` in Intellij Idea to autoformat?

File: src/main/java/seedu/address/model/deck/Deck.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            return true;
+        }
+
+        if (!(other instanceof Deck)) {
+            return false;
+        }
+
+        Deck otherDeck = (Deck) other;
+        return otherDeck.getDeckName().equals(getDeckName());
+    }
```
Comment (geoboom) [inline src/main/java/seedu/address/model/deck/Deck.java]: Could we just do this here?
```java
        return other == this // short circuit if same object
                || (other instanceof Deck // instanceof handles nulls
                && getDeckName().equals(((Deck) other).getDeckName())); // state check
```

File: src/main/java/seedu/address/model/deck/DeckName.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    @Override
+    public String toString() {
+        return name;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof DeckName // instanceof handles nulls
+                && name.equals(((DeckName) other).name)); // state check
```
Comment (geoboom) [inline src/main/java/seedu/address/model/deck/DeckName.java:39]: I think there should be a getter method `getName()` to be consistent with `Deck.java` above.

**Review by GabrielSimbingyang — APPROVED** [review]
Looks good to me

File: src/main/java/seedu/address/model/deck/Deck.java (hunk 1 above)
Comment (MelanieNPS) [reply src/main/java/seedu/address/model/deck/Deck.java:71]: The indentation doesn't change, seems to be that way. In other classes like Entry.java as well

Comment (geoboom) [reply src/main/java/seedu/address/model/deck/Deck.java:71]: gotcha

**Comment by geoboom** [comment]
lgtm

File: src/main/java/seedu/address/model/deck/Deck.java (hunk 2 above)
Comment (MelanieNPS) [reply src/main/java/seedu/address/model/deck/Deck.java]: Yep changed. Ty :)

**Pull request merged**